### DISCUSSION TITLE
Desktop toolbar is showing system folders on  the Taskbar. How do I remove them?

### DISCUSSION THREAD

**Discussion by JonahWick** [valinet/ExplorerPatcher#1418] [Q&A]
I've enabled the Desktop toolbar function for the Taskbar, but it shows a bunch of extra folder icons that are not on my desktop. Right clicking these icons doesn't offer an option to remove them from the toolbar. Is there a way to remove them?
![Screenshot 2023-05-04 005534](https://user-images.githubusercontent.com/132539412/236115615-17ff31b2-fd51-4033-ba96-e7f0e6c9e6e9.jpg)

**Comment by cushlomockree** [comment]
Windows 11 Pro, 22H2, Bld.: 22621.1702
Same thing here. It is displaying icons for many but not all folders in  %HOMEDRIVE%\\%HOMEPATH% (Usually C:\Users\user_name) instead of  %HOMEDRIVE%\\%HOMEPATH%\Desktop (Usually C:\Users\user_name\Desktop).

**Comment by pyrates999** [comment]
Is this a feature in explorer patcher that you enabled?  Or did you just right click on the taskbar and go to toolbars and clicked on desktop?

If it is not an option in explorer patcher, uninstall explorer patcher, reboot, and enable it like you did before.  And let us know if it shows up or not like this.

**Comment by Tome-Wyrm** [comment]
![image](https://github.com/valinet/ExplorerPatcher/assets/146680834/844b6b43-8355-479b-99b4-ae88b81839cc)

I did simply right click the taskbar and enable "desktop"

![image](https://github.com/valinet/ExplorerPatcher/assets/146680834/915bf283-a4f0-45b0-862c-4238bf21846a)

I would like to know how to de-clutter this toolbar as well.

ExplorerPatcher version: 22621.3527.65.1
Windows version: 22631.3447

I have been having this problem for months, it just finally annoyed me enough to want to fix it rather than deal with it.

**Comment by pyrates999** [comment]
I don't think there's options for the desktop toolbar as it displays what is on your desktop.

**Comment by Tome-Wyrm** [comment]
Uh... minor problem being:
![image](https://github.com/valinet/ExplorerPatcher/assets/146680834/995eb9f7-b6b0-48cb-851b-1c163b44cb7c)

I don't think even half of those are OPTIONS to appear on the desktop. I know the Desktop library isn't.

What it looks like it is doing is pulling some of the "library" folders from the /Users/<username> folder... don't know why

**Reply by pyrates999** [reply]
Looks like it is assuming you enabled all the options here: https://support.microsoft.com/en-us/windows/show-desktop-icons-in-windows-c13270f0-3812-c71d-f27e-29aa32588b20#ID0EBD=Windows_11

And any other folders you have in library as well.

**Reply by Tome-Wyrm** [reply]
Quite so... which is very odd. The normal desktop icons are one thing... the libraries another.

Some of the libraries cause the toolbar to just close when I hover over them, though unreliably

**Comment by pyrates999** [comment]
Ok, instead of clicking on Desktop in Toolbars, click on new toolbar... and select whatever folder you want it to show the files/folders inside it.

**Comment by Tome-Wyrm** [comment]
Well... that's almost what I want...

Removes the libraries, but also the desktop icons.

I can at least work with this. Store all my "desktop" files and folders in a folder somewhere and have two toolbars.

Thank you!

**Comment by pyrates999** [comment]
Please mark my response as the answer then.

**Comment by Tome-Wyrm** [comment]
I can't, I'm not the original poster

**Reply by pyrates999** [reply]
Oh right.  Ok.

**Comment by pyrates999** [comment]
@JonahWick please check my answer.

**Comment by Zhurama** [comment]
Salutations, 

I have finally found the answer in Win11 24H2 for removing most of these extraneous folders in the "Desktop" toolbar of the classic Win 10 taskbar (Explorer Patcher Version). It involves a few changes t o the rregistry in order to hide them, by adding a hide command to each section. The effect is immediate, so there is no need to either restart the file explorer or the PC.

The path in Regedit is as follows:

HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace

The folders below Namespace contain the listings.

Create a new txt file and name it whatever you like, and then change the file extension from .txt to .reg. Apply the patch and POOF!!! folders are gone. Change the "HiddenByDefault"=dword:00000001 dword value to a "0" if you want them to show again and reapply the patch.

I still cannot find the keys to remove Recycle Bin, and the Network Folder Hide causes Network to also be removed from the left hand side folder tree of The Explorer, so use that with caution.. If any of you find The Network folder listing that does not also remove it from Explorer, then please post.. I have sorely missed having a clean menu and that is the last item.

TIA & Namaste,
Zhur.


```
Windows Registry Editor Version 5.00

; Hide Control Panel Menu from Desktop Toolbar
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{26EE0668-A00A-44D7-9371-BEB064C98683}]
@="ControlPanelHome"
"HiddenByDefault"=dword:00000001

; Hide Control Panel Link from Desktop Toolbar
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{5399E694-6CE5-4D6C-8FCE-1D8870FDCBA0}]
@="ControlPanelStartupPage"
"HiddenByDefault"=dword:00000001

; Hide Desktop Folder from Desktop Toolbar
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{B4BFCC3A-DB2C-424C-B029-7FE99A87C641}]
@="CLSID_ThisPCDesktopRegFolder"
"HiddenByDefault"=dword:00000001

; Hide Documents Folder fr5om Desktop Toolbar
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{A8CDFF1C-4878-43be-B5FD-F8091C1C60D0}]
@="CLSID_ThisPCDocumentsRegFolder"
"HiddenByDefault"=dword:00000001

; Hide My Music from Desktop Toolbar
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{1CF1260C-4DD0-4ebb-811F-33C572699FDE}]
@="CLSID_ThisPCMyMusicRegFolder"
"HiddenByDefault"=dword:00000001

; Hide My Music from Desktop Toollbar
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{3ADD1653-EB32-4cb0-BBD7-DFA0ABB5ACCA}]
@="CLSID_ThisPCMyPicturesRegFolder"
"HiddenByDefault"=dword:00000001

; Hide Videos From Desktop Toolbar
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{A0953C92-50DC-43bf-BE83-3742FED03C9C}]
@="CLSID_ThisPCMyVideosRegFolder"
"HiddenByDefault"=dword:00000001

; Hide Users Folder from Desktop Toolbar and Root Menu of File Explorer
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{59031a47-3f72-44a7-89c5-5595fe6b30ee}]
@="UsersFiles"
"HiddenByDefault"=dword:00000001

; Hide Gallery from Desktop Toolbar and File Explorer
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{e88865ea-0e1c-4e20-9aa6-edcd0212c87c}]
@="{e88865ea-0e1c-4e20-9aa6-edcd0212c87c}"
"HiddenByDefault"=dword:00000001

; Hide Linux Subsystem Folder from Desktop Toolbar
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{B2B4A4D1-2754-4140-A2EB-9A76D9D7CDC6}]
@="Linux"
"HiddenByDefault"=dword:00000001

; Hide Network Folder from Desktop Toolbar
; (This setting also hides Network inside of This PC - Use with caution)
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{F02C1A0D-BE21-4350-88B0-7367FC96EF3C}]
@="Computers and Devices"
"HiddenByDefault"=dword:00000001

; Hide Libraries Folder from Desktop Toolbar
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\{031E4825-7B94-4dc3-B131-E946B44C8DD5}]
@="UsersLibraries"
"HiddenByDefault"=dword:00000001
